lib/stormpath-sdk/http/http_client_request_executor.rb

```diff
@@ -19,18 +19,17 @@ class HttpClientRequestExecutor
       include Stormpath::Http::Authc
       include Stormpath::Util::Assert
 
-      def initialize(api_key, options = {})
+      def initialize(options = {})
         @signer = Sauthc1Signer.new
-        @api_key = api_key
         @http_client = HTTPClient.new options[:proxy]
       end
 
-      def execute_request(request, redirects_limit = 10)
+      def execute_request(request, api_key, redirects_limit = 10)
```

Comment:
Hi, Tnx a lot for the feedback. To avoid confusion, let me see if I understood you correctly.

The method execute_request on http_client_request_executor is called from the data_store class lines 122-124

```
request = Request.new(http_method, href, query, Hash.new, body)
apply_default_request_headers request
response = @request_executor.execute_request request, @api_key
```

The request object is instance of Request class. So we should move the api_key to request object like this

```
request = Request.new(http_method, href, query, Hash.new, body, @api_key)
```

and then we don't need to pass it as an additional argument. In this case api_key will be present on data_store class and request class. Of course this would require some changes to HttpClientRequestExecutor and Sauthc1Signer since the api_key is not passed as an argument. Is this ok?